### DISCUSSION TITLE
Impossible Colors?!?!

### DISCUSSION THREAD

**Discussion by blue-j** [colour-science/colour#1161] [Q&A] [answered]
In converting a wavelength of 480.5 to various RGBs via XYZ, I noted that when I went to encode linear RGB [-0.16635786  0.21035504  0.78802606] but the negative R value was not acceptable to:

`/models/rgb/transfer_functions/gamma.py:96: RuntimeWarning: invalid value encountered in power
  return as_float(a**exponent)
`

The command I used was:

`encoded_rgb = np.array(colour.RGB_COLOURSPACES['Adobe RGB (1998)'].cctf_encoding(linear_rgb))`

This inspired me to ask the question about impossible colors.  Should we allow processing and report the negative values?  Lindbloom opts for that.   I am not a professional color scientist and am unsure about which direction to go for this.  Right now I am thinking that if there is a negative value, it cannot be represented in the target space, and we should omit it.

- J

**Comment by blue-j** [comment]
I've decided to show the negative values as useful information, but I am not sure how to get around the encoding function.  Maybe just report linear RGB for those?  clamping to zero is misinformation in my view.

**Comment by thomasmansencal** [comment, marked as answer]
I would recommend preserving negative colour values as how long you can, a negative sRGB colour might be a positive BT.2020 or DCI-P3 one.
